From e9f241c890ee8cd7188686fe2eb01b516787e9e1 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:38 -0400
Subject: [tools] lib/traceevent: Unregister handler when hrtimer plugin is unloaded

Message-id: <1408953210-26343-271-git-send-email-jolsa@redhat.com>
Patchwork-id: 89032
O-Subject: [PATCH RHEL7.1 BZ1133083 270/322] tools lib traceevent: Unregister handler when hrtimer plugin is unloaded
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 4061edaa54744dca833051119e763f073dd3c334
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Jan 16 11:31:10 2014 +0900

description
===========
The timer handlers should be unregistered when the plugin is unloaded
otherwise they'll try to access invalid memory.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/plugin_hrtimer.c b/tools/lib/traceevent/plugin_hrtimer.c
index 0b0ebf3..12bf14c 100644
--- a/tools/lib/traceevent/plugin_hrtimer.c
+++ b/tools/lib/traceevent/plugin_hrtimer.c
@@ -76,3 +76,13 @@ int PEVENT_PLUGIN_LOADER(struct pevent *pevent)
           timer_start_handler, NULL);
  return 0;
 }
+
+void PEVENT_PLUGIN_UNLOADER(struct pevent *pevent)
+{
+ pevent_unregister_event_handler(pevent, -1,
+     "timer", "hrtimer_expire_entry",
+     timer_expire_handler, NULL);
+
+ pevent_unregister_event_handler(pevent, -1, "timer", "hrtimer_start",
+     timer_start_handler, NULL);
+}
-- 
1.7.1